sys/kern/uipc_mbuf.c

```diff
@@ -410,7 +411,11 @@ m_pkthdr_init(struct mbuf *m, int how)
 #ifdef MAC
 	int error;
 #endif
-	m->m_data = cheri_kern_setbounds(m->m_pktdat, MHLEN);
+	m->m_data = m->m_pktdat;
+#ifdef __CHERI_SUBOBJECT_BOUNDS__
```

Comment:
I wonder whether we should have an assertion macro for this, e.g. `KASSERT_SUBOBJECT()` or something like that.

Reply:
That seems reasonable.

Reply:
Done.